WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/WooShipping Create Shipping Labels/WooShipping Package and Rate Selection/WooShippingAddPackageView.swift

```diff
@@ -64,100 +92,199 @@ struct WooShippingAddPackageView: View {
                     } label: {
-                        Text(option.name)
-                            .bodyStyle()
-                        if packageType == option {
-                            Image(uiImage: .checkmarkStyledImage)
+                        HStack {
+                            Text(packageType.name)
+                                .bodyStyle()
+                            Spacer()
+                            Image(systemName: "chevron.up.chevron.down")
+                        }
+                        .padding()
+                    }
+                    .roundedBorder(cornerRadius: 8, lineColor: Color(.separator), lineWidth: 1)
+                    AdaptiveStack(spacing: 8) {
+                        ForEach(WooShippingAddPackageDimensionView.DimensionType.allCases, id: \.self) { dimensionType in
+                            WooShippingAddPackageDimensionView(dimensionType: dimensionType, fieldValue: Binding(get: {
+                                return self.customPackageViewModel.fieldValues[dimensionType] ?? ""
+                            }, set: { value in
+                                self.customPackageViewModel.fieldValues[dimensionType] = value
+                            }), focusedField: _focusedField)
+                        }
+                    }
+                    .toolbar {
+                        ToolbarItemGroup(placement: .keyboard) {
+                            Group {
+                                Button(action: {
+                                    onBackwardButtonTapped()
+                                }, label: {
+                                    Image(systemName: "chevron.backward")
+                                })
+                                .disabled(focusedField == WooShippingAddPackageDimensionView.DimensionType.allCases.first)
+                                Button(action: {
+                                    onForwardButtonTapped()
+                                }, label: {
+                                    Image(systemName: "chevron.forward")
+                                })
+                                .disabled(focusedField == WooShippingAddPackageDimensionView.DimensionType.allCases.last)
+                                Spacer()
+                                Button {
+                                    dismissKeyboard()
+                                } label: {
+                                    Text(Localization.keyboardDoneButton)
+                                        .bold()
+                                }
+                            }
+                            .renderedIf(focusedField != nil)
                         }
                     }
+                    Toggle(isOn: $showSaveTemplate) {
+                        Text(Localization.saveNewPackageTemplate)
+                            .font(.subheadline)
+                    }
+                    .tint(Color(.withColorStudio(.wooCommercePurple, shade: .shade60)))
```

Comment:
I'd suggest using the semantic color `Color(.primary)` or `Color(.accent)` here (and elsewhere in this view). That makes it easier to adjust later if we update the app colors, and it adjusts the color in dark mode to fit our usual dark mode theme.